### PR TITLE
08 - Internationalisation

### PR DESCRIPTION
**Pull request by mar7in91Wolox**
# Summary
- I added the .json of three languages (es-AR, fr-FR and en-US)
- **REMEMBER**: the folder _Resources_ must be in _src_ folder.

# Trello Card
https://trello.com/c/bJ33Sxor

# Screenshots
image
image
image

### CODE REVIEW

File: training-net/src/Controllers/HomeController.cs [hunk 1]
```diff
@@ -5,14 +5,21 @@
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Mvc;
 using training_net.Models;
```
[the remark this answers is not in the document]
Comment (mar7in91Wolox) [reply training-net/src/Controllers/HomeController.cs]: ok!|

File: training-net/src/Resources/en-US.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+    "HelloWorldButton": "Hello World!",
+    "HelloWorldTitle": "This title was generated by the CONTROLLER",
+    "HelloWorldParagraph": "...and this paragraph too..."
+}
```
[the remark this answers is not in the document]
Comment (mar7in91Wolox) [reply training-net/src/Resources/en-US.json]: done

File: training-net/src/Resources/es-AR.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+    "HelloWorldButton": "Hola Mundo!",
+    "HelloWorldTitle": "Este título es generado por el CONTROLADOR",
+    "HelloWorldParagraph": "...y este párrafo también..."
+}
```
[the remark this answers is not in the document]
Comment (mar7in91Wolox) [reply training-net/src/Resources/es-AR.json]: done

File: training-net/src/training-net.csproj [hunk 4]
```diff
@@ -3,8 +3,9 @@
...
-    <PackageReference Include="Microsoft.AspNetCore.App" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="2.1.1" />
-    <PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="2.1.1.1" />
+    <PackageReference Include="Microsoft.AspNetCore.App"/>
+    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="2.1.1"/>
+    <PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="2.1.1.1"/>
+    <PackageReference Include="LocalizationCultureCore" Version="1.1.2"/>
   </ItemGroup>
-</Project>
+</Project>
```
[the remark this answers is not in the document]
Comment (mar7in91Wolox) [reply training-net/src/training-net.csproj]: done

File: training-net/src/Startup.cs [hunk 5]
```diff
@@ -29,9 +32,8 @@ public void ConfigureServices(IServiceCollection services)
                 options.CheckConsentNeeded = context => true;
                 options.MinimumSameSitePolicy = SameSiteMode.None;
             });
-
-
             services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_1);
+            CultureInfo.CurrentUICulture = new CultureInfo("es-AR");
```
[the remark this answers is not in the document]
Comment (mar7in91Wolox) [reply training-net/src/Startup.cs]: done!

File: training-net/src/Controllers/HomeController.cs [hunk 6]
```diff
@@ -1,16 +1,26 @@
...
 using Microsoft.AspNetCore.Mvc;
 using training_net.Models;
+using Microsoft.AspNetCore.Mvc.Localization;
 
 namespace training_net.Controllers
 {
     public class HomeController : Controller
     {
-        public IActionResult Index() => View();
+      /*   public HomeController(IHtmlLocalizer<HomeController> localizer)
```
Comment (truccomarcos) [inline training-net/src/Controllers/HomeController.cs]: Remember to remove the comments on this file.
It's okay to have them if we are clarifying something, or if the code will be used, but in this case they aren't needed and might clutter the file once more functionalities are added.

File: training-net/src/Controllers/HelloWorldController.cs [hunk 7]
```diff
@@ -1,13 +1,20 @@
+using System.Globalization;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Localization;
 
 namespace training_net.Controllers
 {
     public class HelloWorldController : Controller
     {
+        private readonly IHtmlLocalizer<HelloWorldController> _localizer;
```
Comment (truccomarcos) [inline training-net/src/Controllers/HelloWorldController.cs]: Don't forget to separate the methods with blank lines (we usually keep attributes or method definitions in interfaces together, having them separated here improves readability).

File: training-net/src/Controllers/HelloWorldController.cs [hunk 8]
```diff
@@ -1,13 +1,20 @@
...
+        public HelloWorldController(IHtmlLocalizer<HelloWorldController> localizer)
+        {
+            this._localizer = localizer;
+        }
         public IActionResult Index()
         {
-            ViewData["firstTitle"] = "This title was generated by the CONTROLLER";
-            ViewData["firstParagraph"] = "...and this paragraph too...";
+            ViewData["Title"] = _localizer["HelloWorldTitle"].Value;
```
Comment (truccomarcos) [inline training-net/src/Controllers/HelloWorldController.cs]: Try using the localizer as a property (same way we did we UnitOfWork)

Comment (mar7in91Wolox) [reply training-net/src/Controllers/HelloWorldController.cs]: Done

File: training-net/src/Controllers/HelloWorldController.cs (hunk 7 above)
Comment (mar7in91Wolox) [reply training-net/src/Controllers/HelloWorldController.cs]: done

**Pull request merged**